cnf-certification-test/lifecycle/podsets/podsets.go

```diff
@@ -40,6 +40,8 @@ var WaitForDeploymentSetReady = func(ns, name string, timeout time.Duration) boo
 		if err == nil && IsDeploymentReady(dp) {
 			logrus.Tracef("%s is ready", provider.DeploymentToString(dp))
 			return true
+		} else if dp == nil {
+			logrus.Error("error: deployment object found to be nil")
 		} else if err != nil {
 			logrus.Errorf("Error while getting the %s, err: %s", provider.DeploymentToString(dp), err)
```

Comment:
Suggestion: dp will be nil only when err != nil, so instead of adding a new "else if" branch, you can just use "ns" & "name" here from this function's params:
```
  } else if err != nil {
    logrus.Errorf("Error while getting deployment %s (ns: %s), err: %s", name, ns, err)
  }
```

Reply:
I think the check for a `dp != nil`/`err != nil` should happen right after the call to `provider.GetUpdatedDeployment()` before  `dp` is used in `IsDeploymentReady()`

Reply:
Fixed.  Added string recommended by @greyerof